### PR TITLE
[autodetect] add pointer for custom user data

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

File: include/freerdp/autodetect.h [hunk 1]
```diff
@@ -28,6 +28,16 @@
...
 {
 #endif
+
+	typedef enum
+	{
+		AUTODETECT_STATE_INITIAL,
+		AUTODETECT_STATE_REQUEST,
+		AUTODETECT_STATE_RESPONSE,
+		AUTODETECT_STATE_COMPLETE,
+		AUTODETECT_STATE_FAIL
```
Comment (pnowack) [inline include/freerdp/autodetect.h:38]: Nit: Could also end with a `,` here. Then, in case of state additions, we won't have change unrelated existing lines.

**Review by pnowack — APPROVED** [review]

File: include/freerdp/autodetect.h (hunk 1 above)
Comment (akallabeth) [reply include/freerdp/autodetect.h:38]: that would produce quite a few warnings during build as gcc/clang complain about that when set to pedantic/everything ;)

**Review by mfleisz — APPROVED** [review]
